docs/utils/json.md

```diff
@@ -0,0 +1,422 @@
+Thunder uses its own JSON parser for serialising and deserialising JSON objects. The parser is built around the idea of strongly-typed JSON objects, where every JSON document/object has a corresponding C++ class that represents it. This design has a few advantages:
+
+* Increased parsing performance - the parser can quickly discard parts of the document that are not relevant and the data types are known ahead of time
+* Compile-time checks - each JSON document is a strongly typed object so the types of all fields are known by the compiler. This removes the chance to make a mistake when accessing document fields at runtime.
+
+
+The only downside of this approach it that all JSON documents must have a known structure at compile time. However, for embedded systems this is considered an acceptable trade-off.
+
+## Define a JSON document
+
+All JSON documents should be defined as instances of `Core::JSON::Container`.
+
+The class will contains public variables for each JSON property that should be accessible. In the constructor of the class, the `Add()` method should be called for each property to map the JSON object name to the variable. The class initialiser list should define sensible default values for each variable.
```

Comment:
"The class will contains public variables" -> "The class contains public variables" or "The class will contain public variables" ??